### PR TITLE
Convert config.ini to config.yml

### PR DESCRIPTION
**Pull request by chenslucky**
# Description

**Local Tests:**
1. with split:
image

2. no split:
image

**Cloud Tests:**
1. with split:  https://ml.azure.com/runs/zen_sand_ns4ljnvv9y?wsid=/subscriptions/96aede12-2f73-41cb-b983-6d11a904839b/resourcegroups/promptflow/workspaces/yaopfeus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47
2. no split:  https://ml.azure.com/runs/mango_table_6ysb8331l0?wsid=/subscriptions/96aede12-2f73-41cb-b983-6d11a904839b/resourcegroups/promptflow/workspaces/yaopfeus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: examples/gen_test_data/gen_test_data/run.py [hunk 1]
```diff
@@ -1,15 +1,15 @@
+import argparse
 import json
 import os
 import shutil
 import time
 from datetime import datetime
 from pathlib import Path
 
-import configargparse
```
Comment (chjinche) [inline examples/gen_test_data/gen_test_data/run.py:8]: remove `configargparse` from `requirements.txt`

File: examples/gen_test_data/gen_test_data/run.py [hunk 2]
```diff
@@ -224,67 +224,34 @@ def get_ml_client(subscription_id: str, resource_group: str, workspace_name: str
...
-    copied_flow_folder = args.flow_folder + "_" + time.strftime("%b-%d-%Y-%H-%M-%S") + "_temp"
-
+    copied_flow_folder = config["flow_folder"] + "_" + time.strftime("%b-%d-%Y-%H-%M-%S") + "_temp"
     try:
         should_skip_split_documents = False
-        document_nodes_file = convert_to_abs_path(args.document_nodes_file)
-        documents_folder = convert_to_abs_path(args.documents_folder)
-        flow_folder = convert_to_abs_path(args.flow_folder)
-        output_folder = convert_to_abs_path(args.output_folder)
+        document_nodes_file = convert_to_abs_path(config.get("document_nodes_file", None))
```
Comment (chjinche) [inline examples/gen_test_data/gen_test_data/run.py:243]: better to be `None`->''?

File: examples/gen_test_data/gen_test_data/run.py [hunk 3]
```diff
@@ -224,67 +224,34 @@ def get_ml_client(subscription_id: str, resource_group: str, workspace_name: str
...
+        documents_folder = convert_to_abs_path(config.get("documents_folder", None))
+        flow_folder = convert_to_abs_path(config.get("flow_folder", "./"))
```
Comment (chjinche) [inline examples/gen_test_data/gen_test_data/run.py:245]: Does this work both windows and linux?

Comment (chenslucky) [reply examples/gen_test_data/gen_test_data/run.py]: update to None

File: examples/gen_test_data/config.yml.example [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    model_or_deployment_name: "<your-model-or-deployment-name>"
+  generate_suggested_answer:
+    connection: "<your-connection-name>"
+    model_or_deployment_name: "<your-model-or-deployment-name>"
+  validate_suggested_answer:
+    connection: "<your-connection-name>"
+    model_or_deployment_name: "<your-model-or-deployment-name>"
+
+
+# Local section: this section is for local test data generation related configuration.
```
Comment (chjinche) [inline examples/gen_test_data/config.yml.example:32]: Can skip if not run in local.

File: examples/gen_test_data/config.yml.example [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+output_folder: "<your-output-folder-path>"
+flow_batch_run_size: 4
+
+
+# Cloud section: this section is for cloud test data generation related configuration.
```
Comment (chjinche) [inline examples/gen_test_data/config.yml.example:37]: Can skip if not run in cloud

File: examples/gen_test_data/config.yml.example [hunk 6]
```diff
@@ -0,0 +1,49 @@
+# Common section: this section provides common values for all other sections.
```
Comment (chjinche) [inline examples/gen_test_data/config.yml.example:1]: Required.

File: examples/gen_test_data/config.yml.example (hunk 4 above)
Comment (chjinche) [inline examples/gen_test_data/config.yml.example:32]: Can skip if not run in local.

File: examples/gen_test_data/config.yml.example (hunk 5 above)
Comment (chjinche) [inline examples/gen_test_data/config.yml.example:37]: Can skip if not run in cloud.

**Review by chjinche — APPROVED** [review]

**Pull request merged**